pkg/controller/reconciler/deployment_reconciler.go

```diff
@@ -402,10 +402,16 @@ func (r *DeploymentReconciler) startPutOrDeleteOperationIfNeeded(ctx context.Con
 		poller, err := deleteContainer(ctx, r.Radius, annotations.Status.Container)
 		if err != nil {
 			return nil, nil, false, err
+		} else if poller != nil {
+			return nil, poller, false, nil
 		}
 
-		return nil, poller, false, err
-	} else if !annotations.IsUpToDate() {
+		// Deletion completed synchronously.
+		annotations.Status.Container = ""
+	}
```

Comment:
All of the code that handles pollers is changing like this. We need to handle the case where the operation completed synchronously and successfully.